extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -487,6 +493,12 @@ export class AmpA4A extends AMP.BaseElement {
               creativeParts.creative) {
             this.creativeBody_ = creativeParts.creative;
           }
+          if (creativeParts && creativeParts.size
```

Comment:
Actually, we check if creativeParts is null thrice in this then() block, and I'm not sure if it's avoidable. We could have a block like you wrote 

```
if (!creativeParts) {
  return Promise.resolve();
}
```

at the top of this then() block, which would save us the trouble of checking for creativeParts' existence, and then another like

```
if (!creativeParts.signature) {
  return Promise.resolve();
}
```

after my if block to bail out in the case of no signature, but that seems messier than its worth. What do you think?